core/hotspot/rule_manager.go

```diff
@@ -72,6 +73,8 @@ func getTrafficControllersFor(res string) []TrafficShapingController {
 // bool: indicates whether the internal map has been changed;
 // error: indicates whether occurs the error.
 func LoadRules(rules []*Rule) (bool, error) {
+	updateRuleMux.Lock()
+	defer updateRuleMux.Unlock()
 	tcMux.RLock()
```

Comment:
Perhaps tcMux.Rlock is not necessary; currentRules can be locked with updateRuleMux

Reply:
Done, thanks.